lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -604,6 +595,27 @@ void IMConstPropPass::visitStrictConnect(StrictConnectOp connect) {
       << "strictconnect destination is here";
 }
 
+void IMConstPropPass::visitRegResetOp(RegResetOp regReset) {
+  // If the reg has a non-ground type, then it is too complex for us to handle,
+  // mark it as overdefined.
+  // TODO: Eventually add a field-sensitive model.
+  if (!regReset.getType().getPassiveType().isGround())
+    return markOverdefined(regReset);
+
+  // The reset value may be known - if so, merge it in if the enable is greater
+  // than invalid.
+  auto srcValue = getExtendedLatticeValue(regReset.resetValue(),
+                                          regReset.getType().cast<FIRRTLType>(),
+                                          /*allowTruncation=*/true);
+  auto enable = getExtendedLatticeValue(
+      regReset.resetSignal(),
+      regReset.resetSignal().getType().cast<FIRRTLType>(),
+      /*allowTruncation=*/true);
+  if (enable.isOverdefined() ||
+      (enable.isConstant() && !enable.getConstant().getValue().isZero()))
+    mergeLatticeValue(regReset, srcValue);
```

Comment:
Not directly related to this PR, is it correct to merge `srcValue` if reset signal is overdefined? For example, currently `reg` is folded into constant 1 by IMCP but `r` might be always 0 in the caller side. 
```scala
circuit Foo:
  module Foo:
    input clock: Clock
    input r: UInt<1>
    output o: UInt<1>

    reg reg : UInt<1>, clock with :
      reset => (r, UInt<1>(1))
    o <= reg
```

Reply:
It should be fine.  if reg is assigned at all, then reg will be overdefined.  otherwise, we are just saying that instead of undefined value at power-on, we see the reset value at power-on prior to a reset.
It's not behavior preserving, but it is reasonable semantics for a register.

Reply:
Thanks, that makes sense to me!